jetty-client/src/main/java/org/eclipse/jetty/client/AbstractConnectionPool.java

```diff
@@ -98,16 +97,41 @@ public boolean isClosed()
 
     @Override
     public Connection acquire()
+    {
+        return acquire(true);
+    }
+
+    /**
+     * <p>Returns an idle connection, if available;
+     * if an idle connection is not available, and the given {@code create} parameter is {@code true},
+     * then schedules the opening of a new connection;
+     * otherwise returns {@code null}.</p>
+     *
+     * @param create whether to schedule the opening of a connection if no idle connections are available
+     * @return an idle connection or {@code null} if no idle connections are available
+     */
+    protected Connection acquire(boolean create)
```

Comment:
The documented contract is not respected by the `RoundRobinConnectionPool` subclass. That should probably be documented too.

Reply:
Documented.